lib/URI/_userpass.pm

```diff
@@ -40,8 +40,8 @@ sub password
 	my $user = defined($info) ? $info : "";
 	$user =~ s/:.*//;
 
-	if (!defined($new) && !length($user)) {
-	    $self->userinfo(undef);
+	if (!defined($new)) {
+	    $self->userinfo($user || undef);
```

Comment:
length() is significant to protect username `0`.

Reply:
@joenio are you able to address the comment above?